client/src/lib/solana/solid-data.ts

```diff
@@ -10,17 +10,43 @@ import {
 } from 'did-resolver';
 
 export class SolidData extends Assignable {
+  // derived
+  account: SolidPublicKey;
+  cluster: ClusterType;
+
+  // persisted
+  authority: SolidPublicKey;
   context: string[];
-  did: DecentralizedIdentifier;
   verificationMethod: VerificationMethod[];
-  authentication: DecentralizedIdentifier[];
-  capabilityInvocation: DecentralizedIdentifier[];
-  capabilityDelegation: DecentralizedIdentifier[];
-  keyAgreement: DecentralizedIdentifier[];
-  assertionMethod: DecentralizedIdentifier[];
+  authentication: string[];
+  capabilityInvocation: string[];
+  capabilityDelegation: string[];
+  keyAgreement: string[];
+  assertionMethod: string[];
   service: ServiceEndpoint[];
 
-  merge(other: SolidData, overwriteArrays: boolean = false): SolidData {
+  static fromAccount(
+    accountKey: PublicKey,
+    accountData: Buffer,
+    cluster: ClusterType
+  ): SolidData {
+    const solidData = SolidData.decode<SolidData>(accountData);
```

Comment:
Hm, I wonder if we can eventually do better than this?

Reply:
yeah I played a little bit with fixing the return value for decode() but didn't want to over-complicate this PR. It definitely needs fixing as well as the constructors for Assignable which are not typesafe, but I will do that in another PR